scipy/_lib/_array_api.py

```diff
@@ -545,3 +551,52 @@ def xp_sign(x: Array, /, *, xp: ModuleType | None = None) -> Array:
     sign = xp.where(x < 0, -one, sign)
     sign = xp.where(x == 0, 0*one, sign)
     return sign
+
+
+_SCIPY_ARRAY_API = os.environ.get("SCIPY_ARRAY_API", False)
+array_api_compat_prefix = "scipy._lib.array_api_compat"
+
+
+def get_array_subpackage_func(f_name, xp, n_array_args, root_namespace,
+                              subpackage, generic_implementations):
+    spx = scipy_namespace_for(xp)
+    f = None
+    if is_numpy(xp):
+        f = getattr(root_namespace, f_name, None)
+    elif spx is not None:
+        f = getattr(getattr(spx, subpackage, None), f_name, None)
+        f = f or getattr(getattr(xp, subpackage, None), f_name, None)
```

Comment:
CuPy's `linalg` is in `cupy`, but its `signal` is in `cupyx`. Perhaps there is a better way to handle this, but for now, look in one, and if not found, look in the other.